extensions/entsoe/src/XnodeAdder.cpp

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <powsybl/iidm/extensions/entsoe/XnodeAdder.hpp>
+
+#include <powsybl/iidm/DanglingLine.hpp>
+#include <powsybl/iidm/extensions/entsoe/Xnode.hpp>
+#include <powsybl/stdcxx/instanceof.hpp>
+#include <powsybl/stdcxx/make_unique.hpp>
+
+namespace powsybl {
+
+namespace iidm {
+
+namespace extensions {
+
+namespace entsoe {
+
+XnodeAdder::XnodeAdder(Extendable& extendable) :
+    ExtensionAdder(extendable) {
+}
+
+std::unique_ptr<Extension> XnodeAdder::createExtension(Extendable& extendable) {
+    if (stdcxx::isInstanceOf<DanglingLine>(extendable)) {
+        return stdcxx::make_unique<Xnode>(dynamic_cast<DanglingLine&>(extendable), m_code);
```

Comment:
I think it should not be possible anymore to create an Xnode extension without using the adder. So I propose to make the constructor private, and declare the XNodeAdder as a friend class.
@sebalaig What do you think?

I think you have to cleanup the existing code, specially in unit tests to use the new API.

Reply:
We decided to be Java consistent and to enable the creation without using the adder.